flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -207,7 +207,12 @@ convertOptExtentExpr(Fortran::lower::AbstractConverter &converter,
 }
 
 /// Does this expr designate an allocatable or pointer entity ?
-static bool isAllocatableOrPointer(const Fortran::lower::SomeExpr &expr) {
+static bool
+isAllocatableOrPointer(const Fortran::lower::SomeExpr &expr,
+                       Fortran::lower::AbstractConverter &converter) {
+  // First check if this is a pointer function result.
+  if (Fortran::evaluate::IsObjectPointer(expr, converter.getFoldingContext()))
```

Comment:
Why not just pass the `bool` result here instead of the bridge object?

Reply:
I am not sure I follow, do you want to get rid of the `Fortran::lower::AbstractConverter &converter`  argument of isAllocatableOrPointer ? If so it is needed because IsObjectPointer needs a folding context, is what you are suggesting that the callers of `isAllocatableOrPointer ` calls `IsObjectPointer` and pass the result ?

Reply:
It looks like the call can be done in the caller rather than passing the entire bridge object so that the folding context can be used here.

Passing the bridge suggests that this test is doing something really major from the calling side. It isn't. It is just a straightforward test.